corehq/apps/api/odata/tests/utils.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import base64
+
+from django.test.client import Client
+from django.urls import reverse
+
+from corehq.apps.domain.models import Domain
+from corehq.apps.users.models import WebUser
+
+
+class OdataTestMixin(object):
+
+    view_urlname = None
+
+    @classmethod
+    def setUpClass(cls):
+        super(OdataTestMixin, cls).setUpClass()
+        cls.client = Client()
+        cls.domain = Domain(name='test_domain')
+        cls.domain.save()
+        cls.web_user = WebUser.create(cls.domain.name, 'test_user', 'my_password')
+
+    def tearDown(self):
+        super(OdataTestMixin, self).tearDown()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.domain.delete()
+        cls.web_user.delete()
+        super(OdataTestMixin, cls).tearDownClass()
+
+    def test_no_credentials(self):
```

Comment:
independent of where you and @emord land on the mixin discussion it seems odd that a class designed to be inherited would also contain all these actual tests (which would then get run once per subclass). was that by design?

Reply:
it was by design, but this might not be a good idea because swapping the order of the base classes would mess things up.  Probably better to duplicate tests or to explicitly call helper functions that run the assertions.